os/arch/arm/src/imxrt/imxrt_mpuinit.c

```diff
@@ -137,8 +139,21 @@ void imxrt_mpu_initialize(void)
 	int i;
 
 	for (i = 0; i < (sizeof(regions_info) / sizeof(struct mpu_region_info)); i++) {
-		lldbg("Region = %u base = 0x%x size = %u\n", g_mpu_region_nr, regions_info[i].base, regions_info[i].size);
-		regions_info[i].call(g_mpu_region_nr++, regions_info[i].base, regions_info[i].size);
+#ifndef CONFIG_APP_BINARY_SEPARATION
+		switch (i) {
+		case 0:
+			mpu_reg_nr = MPU_REG_NUM_UFLASH;
+			break;
+		case 1:
+			mpu_reg_nr = MPU_REG_NUM_URAM;
+			break;
+		default:
+			mpu_reg_nr = i;
```

Comment:
Default case can be considered as error? because the next mpu region is used by other modules..

Reply:
Modified here as per Kishore's suggestion below